tests/filecheck/transforms/copy_to_dma.mlir

```diff
@@ -0,0 +1,23 @@
+// RUN: ./compiler/snax-opt %s -p snax-copy-to-dma --print-op-generic | filecheck %s
+
+"builtin.module"() ({
+  "func.func"() <{"sym_name" = "simple_mult", "function_type" = (memref<?xi32>, memref<?xi32>) -> (), "sym_visibility" = "public"}> ({
+  ^0(%arg0 : memref<?xi32>, %arg1 : memref<?xi32>):
+    "memref.copy"(%arg0, %arg1) : (memref<?xi32>, memref<?xi32>) -> ()
+    "func.return"() : () -> ()
+  }) : () -> ()
+}) : () -> ()
+
+//CHECK: "builtin.module"() ({
+//CHECK-NEXT:   "func.func"() <{"sym_name" = "simple_mult", "function_type" = (memref<?xi32>, memref<?xi32>) -> (), "sym_visibility" = "public"}> ({
+//CHECK-NEXT:   ^0(%arg0 : memref<?xi32>, %arg1 : memref<?xi32>):
```

Comment:
Does this still work for 2D memrefs?
Should it fail if it encounters such memrefs?

Reply:
with the new check,  it ignores memrefs with more than 1 dimension for now